### PR TITLE
new document upload page for patients

### PR DESCRIPTION
**Pull request by lawyu89**
### Why

New documents page to allow patients to upload documents to their practitioners.

### Ticket

https://app.shortcut.com/capable/story/8277
https://app.shortcut.com/capable/story/8278
partially https://app.shortcut.com/capable/story/8279 (pagination component but need to update task/goals/careplans to use it)

### Accomplishments

- [x] New documents page with all the bells and whistles

loom:
https://www.loom.com/share/a77721ff67bb4d5eb9ca51c2994738ee

### CODE REVIEW

File: src/components/PaginatedList.tsx [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+  currentRecords,
+  sectionHeader,
+  sx,
+  showMore,
+  totalRecords,
+  renderRecord,
+  clearCollection,
+}: {
+  page: number;
+  currentRecords: any[];
```
Comment (lawyu89) [inline src/components/PaginatedList.tsx:17]: intentionally using any feels good 😄

File: src/features/attachments/hooks/useAttachements.ts [hunk 2]
```diff
@@ -0,0 +1,108 @@
+import { useMutation, useQuery, useQueryClient } from "@tanstack/react-query";
+import { ReactQueryKeys } from "constants/keys";
+import { Attachment } from "../models/Attachment";
+import api from "capableApi/index";
```
Comment (danstans) [inline src/features/attachments/hooks/useAttachements.ts:4]: file should be useAttachments without the e

File: src/features/attachments/hooks/useAttachements.ts [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+    ...params,
+  });
+  console.log({
+    ...{
+      page: documentPage,
+      size: 10,
+      byUserId: [patientId],
+      sortBy: ["-created_at"],
+    },
+    ...params,
```
Comment (danstans) [inline src/features/attachments/hooks/useAttachements.ts:23]: 🎮

File: src/features/attachments/hooks/useAttachements.ts (hunk 2 above)
Comment (lawyu89) [reply src/features/attachments/hooks/useAttachements.ts:4]: opsies thanks for catching haha

File: src/utils/strings.tsx [hunk 4]
```diff
@@ -80,10 +80,27 @@ export const convertRetailPriceCentsToRetailPrice = (retailCents: number) => {
...
+export const calculateByteString = (bytes: number) => {
+  if (bytes >= GIGABYTE) {
+    return `${(bytes / GIGABYTE).toFixed(2)} GB`;
+  } else if (bytes >= MEGABYTE) {
+    return `${(bytes / MEGABYTE).toFixed(2)} MB`;
+  } else if (bytes >= KILOBYTE) {
+    return `${(bytes / KILOBYTE).toFixed(2)} KB`;
+  } else {
+    return `${bytes} bytes`;
+  }
```
Comment (danstans) [inline src/utils/strings.tsx:105]: case for when not bytes? (catching the string concatenation that uses this function)

Comment (lawyu89) [reply src/utils/strings.tsx:105]: this function accepts bytes (as a number) so it there should never be a case when it isnt bytes.

File: src/pages/PaymentSuccess.tsx [hunk 5]
```diff
@@ -2,23 +2,22 @@ import { useState, useEffect } from "react";
...
+  const [subscription, setSubscription] = useState<Subscription | undefined>();
   const { isLoading, data: subscriptions } = useActiveSubscription(currentPatient?.id);
   const { subscriptionId } = useParams<{
     subscriptionId?: string;
   }>();
 
   useEffect(() => {
     setSubscription(
-      subscriptions.find((sub: Subscription) => sub.stripe_subscription_id === subscriptionId)
+      subscriptions?.find((sub: Subscription) => sub.stripe_subscription_id === subscriptionId)
```
[the remark this answers is not in the document]
Comment (lawyu89) [reply src/pages/PaymentSuccess.tsx:20]: this is fine (as itll stay undefined if there isnt a subscription).

**Review by LUMacKenn — APPROVED** [review]

**Pull request merged**